LayoutTests/fast/scrolling/mac/keyboard-scrolling-with-mouse-scroll.html

```diff
@@ -11,29 +11,47 @@
             testRunner.dumpAsText();
             testRunner.waitUntilDone();
         }
-        
+
+        let startPosition = 0;
+        let delta = 0;
+        let position = 0;
+        let didScrollPastPosition = false;
+
         async function runTest()
         {
             eventSender.monitorWheelEvents();
             if (!window.testRunner || !testRunner.runUIScript)
                 return;
 
+            document.addEventListener('wheel', (event) => {
+                startPosition = window.scrollY;
+                // debug('startPosition = ' + startPosition);
+                delta = event.deltaY;
+                // debug('delta = ' + delta);
+
+                document.addEventListener('scroll', (event) => {
+                    position = window.scrollY;
+                    // debug('position = ' + position);
+                    if (startPosition + delta != position)
+                        didScrollPastPosition = true;
+                    // shouldBe('startPosition + delta', 'position');
+                    // testRunner.notifyDone();
+                })
+            });
+
             await UIHelper.rawKeyDown("downArrow");
             
             setTimeout(async () => {
                 await UIHelper.rawKeyUp("downArrow");
-                const startingPosition = window.scrollY;
-
-                await UIHelper.mouseWheelScrollAt(0, 0, 0, 0, 0, 0);
+                await UIHelper.mouseWheelScrollAt(0, 0, 0, 0, 0, 1);
 
-                const position = window.scrollY;
-                if (Math.abs(startingPosition - position) < 20)
-                    debug("Successful.");
-                else
-                    debug("Unsuccessful. window.scrollY after wheel event == " + position + "; window.scrollY before wheel event == " + startingPosition);
+                await UIHelper.animationFrame();
+                await UIHelper.animationFrame();
+                await UIHelper.animationFrame();
 
+                shouldBe('didScrollPastPosition', 'false')
                 testRunner.notifyDone();
-            }, 100);
+            }, 300);
```

Comment:
Erg, 300ms test. This is to get it scrolling far? Can you try spacebar (or cmd-down) instead of downarrow, to get it going faster?

Do you need this at all, actually? I would expect it to always fail before this change even with no timeout.

Reply:
It becomes flaky again with 100ms, no timeout, or using the space bar

Reply:
That suggests you have not sorted out all of the races :)